### PR TITLE
Islandora 2060

### PR DESCRIPTION
**Pull request by patdunlavey**
**JIRA Ticket**: [ISLANDORA-2060](https://jira.duraspace.org/browse/ISLANDORA-2060)

# What does this Pull Request do?

### The problem

When content is loaded via an ajax call on a page that has an Openseadragon viewer, the viewer replicates itself inside the `#islandora-openseadragon` div. 

The Drupal.behaviors.islandoraOpenSeadragon:detach function is being called during the ajax call, and this function is disabling but not removing the viewer. It also removes Drupal.IslandoraOpenSeadragonViewer[base]. 

After the detach event is done, a new attach event is triggered. When the attach function is called, it sees that there is no `Drupal.IslandoraOpenSeadragonViewer[base]`, so it adds a new one, thus duplicating the viewer. 

### Desired outcome 

Ajax calls within a page displaying an openseadragon viewer should not cause the viewer to be duplicated.

# What's new?

Adds a condition to the Drupal.behaviors.IslandoraOpenSeadragon.detach function which checks to see if the context of the detach event is the openseadragon viewer. If so, it goes ahead with the detach. But if not, then it does not detach.

# How should this be tested?

* To reproduce the duplication problem, on the page displaying the openseadragon viewer you would need to have a link that makes an ajax call. 
* Once that is implemented, this PR should cause the openseadragon viewer to not be duplicated.

# Additional Notes:

* The detach function would need to function in the case where a module wishes to create and delete openseadragon viewers. For example, if clicking on a search results pops the viewer up in a colorbox, then the close function for that may need to detach the viewer. *This has not been tested!*

# Interested parties
Tag @nigelbanks, @Islandora @7-x-1-x-committers

### CODE REVIEW

**Comment by DiegoPino** [comment]
@patdunlavey @nigelgbanks this one is an improvement and we are close to code freeze. Please give this a push/search for reviewers in the committers group if you want to get this into code hopefully before the 15th! Good luck =)

**Review by nigelgbanks — APPROVED** [review]
This looks good to me, under default Islandora setup the detach function doesn't normally get called so no change in behaviour for 99% of users. In the case of users who have need to perform some Ajax calls on object pages that display the openseadragon viewer this improvement will prevent the issue described.

**Comment by nigelgbanks** [comment]
I'll merge after the standard 24 hours (assuming travis passes etc), unless someone else wants to chime in.

**Comment by DiegoPino** [comment]
That was quick! Travis is still runnig so make sure that passes first. Thanks to both, great response!

**Comment by patdunlavey** [comment]
Thanks for giving this priority, @nigelgbanks !

**Pull request merged**
